### PR TITLE
fix(TDI-38486): tMarketoInput incoming flow

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

See https://jira.talendforge.org/browse/TDI-38486 for full description.

**What is the new behavior?**

tMarketoInput now accepts incoming flow with some restrictions (see JIRA issue).

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/459/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/459/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/460/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/460/)

File: components/components-marketo/src/main/java/org/talend/components/marketo/runtime/MarketoInputWriter.java [hunk 1]
```diff
@@ -13,66 +13,103 @@
...
+        inputRecord = (IndexedRecord) object;
         result.totalCount++;
         result.apiCalls++;
         // This for dynamic which would get schema from the first record
         if (inputSchema == null) {
-            inputSchema = ((IndexedRecord) object).getSchema();
+            inputSchema = inputRecord.getSchema();
+        }
+        if (isDynamic) {
+            adaptSchemaToDynamic();
```
Comment (jzhao-talend) [inline components/components-marketo/src/main/java/org/talend/components/marketo/runtime/MarketoInputWriter.java]: Why call `adaptSchemaToDynamic` for every record ?
I think his would be reduce efficiency.

Comment (undx) [inline components/components-marketo/src/main/java/org/talend/components/marketo/runtime/MarketoInputWriter.java]: Fixed.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/489/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/489/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/490/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/490/)

**Pull request merged**